skops/card/_model_card.py

```diff
@@ -254,13 +297,28 @@ def __init__(
         model_diagram: bool = True,
         metadata: Optional[CardData] = None,
     ) -> None:
-        self.model = model
+        self._model = model
         self.model_diagram = model_diagram
         self._eval_results = {}  # type: ignore
         self._template_sections: dict[str, str] = {}
         self._extra_sections: list[tuple[str, Any]] = []
         self.metadata = metadata or CardData()
 
+    @property
+    def model(self):
+        model = _load_model(self._model)
+        if model is not self._model:
+            self._model = model
+        return model
```

Comment:
I don't think we should change `model`. We should always keep `model` attribute as is given by the user, and only load when we need it in other methods.

Reply:
agreed 👍🏻

Reply:
I'm not sure I fully understand. You want the `model` argument passed by the user to remain unchanged, and when we access it in other methods, we check if it's a string/path and load it? So basically swap around the attribute names `model` and `_model`? Or load the model each time it's accessed (which would be wasteful, since it's used, for instance, in `__repr__` so could be accessed quite frequently).

Reply:
yes, exactly @p-mishra1 .

Reply:
-1 from me, why potentially reload the model multiple times?

Reply:
yes i agree with @BenjaminBossan as there can be multiple calls to `__repr__`.

Reply:
E.g. imagine I create a card in a repl/notebook and call `add` multiple times. Since the method always returns `self`, each call implicitly calls `repr` and thus loads the model.

Reply:
One thing I could see as a compromise would be to call `_load_model` each time but also to cache the model. The problem here is that we cannot use the path to the model as the cache key, we would need to use something like md5sum. And the caching decorators from `functools` don't support custom key functions like this (AFAIK), so we would need to write our own cache or add a dependency.

Reply:
We can load the model everytime in this PR, and in a separate PR fix the caching issue.

Reply:
> We can load the model everytime in this PR, and in a separate PR fix the caching issue.

Okay, let's do it this way and add a caching decorator in a separate PR (or use a [library](https://toolz.readthedocs.io/en/latest/api.html?highlight=memoize#toolz.functoolz.memoize) for it). I guess it's okay to add it later (though I would have liked to have included here) because this functionality was not available beforehand, so it's still a net improvement.

@p-mishra1 Are you okay with that?

Reply:
Yes @BenjaminBossan sounds fine will work on this more, and
just curious do we have slack/forum/discord where we can talk about code design before commiting it, or is it the preferred way.

Reply:
Thanks for continuing the work.

I think ideally these discussions should come up in the PR or the corresponding issue. I thought there was consensus on the suggested approach here, I guess there was some confusion, sorry for that.

Reply:
```python
    def __repr__(self) -> str:

        model = getattr(self, "model", None)
        if model:
            model = _load_model(model) # this line
            model_str = self._strip_blank(repr(model))
            model_repr = aRepr.repr(f"  model={model_str},").strip('"').strip("'")
        else:
            model_repr = None
```

```python
    def _extract_estimator_config(self) -> str:
        """Extracts estimator hyperparameters and renders them into a vertical table.

        Returns
        -------
        str:
            Markdown table of hyperparameters.
        """
        model = _load_model(self.model) # This line
        hyperparameter_dict = model.get_params(deep=True)
```

```python

    def _generate_card(self) -> ModelCard:
        .
        .
        .
        if self.model_diagram is True:
            model = _load_model(self.model) # this line
            model_plot_div = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))

```
so these 3 functions will be changed like this(commented with "this line") while doing lazy loading in card class,
also the test for card __repr__ and __str__ either have to be changed to take card instance and path or the test class have to be duplicated like passing model file as param instead of estimator.
lmk @adrinjalali @BenjaminBossan which method should we prefer.

Reply:
I wonder if this is the right approach or if we should keep the `property` and just do the `_load_model` inside the property. The disadvantage of the suggested approach is that it's very error prone, as soon as new code is added that access the model, it would need to do the same thing. That's why I would prefer using `property`. @adrinjalali wdyt?

Reply:
how about `self.get_model()` which would return the model if `self.model` is a `str`? acceptable middle ground?